mantidproject/mantid
# Updating Job Tree View

**Description of work.**

This PR takes the changes to JobTreeView from #22590 and pulls them out into a separate PR. Changes included here are 
* Added cut functionality to JobTreeView
* Updates to hinting line edit classes to make them exportable through SIP to python
* Updates to the existing Reflectometry GUI to work with the new hinting line edit classes
* The adding of an interface IJobTreeView as well as some tests
* The correction of a bug with SubTrees

**To test:**
The most important test is to check that the existing Reflectometry and SANS Gui's still work. To test the JobTreeView changes requires a GUI updated to use them.


*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
